docs/articles/immutability-in-js.md

```diff
@@ -0,0 +1,108 @@
+# Immutability in JavaScript
+
+According to (wikipédia)[https://en.wikipedia.org/wiki/Immutable_object] an *immutable object (unchangeable object) is an object whose state cannot be modified after it is created*. This rule is quite simple, if you want to modify some property of an object you have to do it on a copy. We will see a bit later what improvements and fancy features it unlocks for our developments.
+
+## EcmaScript
+
+EcmaScript provides utilities to keep our data immutable. APIs of arrays and objects have methods to create copies and even prevent instances to be updated. More recently EcmaScript brings us a brand new syntax that makes it easier to achieve it.
+
+### Object.assign
+
+We want to add a name property in our object.
+
+https://gist.github.com/frinyvonnick/a281f15288d3ab90ce5354f01df52ccf
+
+We can do it with `Object.assign` and a little trick. Basically it copies all properties of an object to another one, thus it mutates the target object. So we use a small trick, by passing an empty object at first parameter we create a new reference.
+
+https://gist.github.com/frinyvonnick/98878e45d6a8ccb4618238dbec175ae9
+
+So we now have a new object with our new name property and a commonNames property remaining unchanged. With this method you can create/overwrite multiple properties at the same time.
```

Comment:
Tu utilises souvent le `So` :smile: Je pense que celui-ci n'est pas indispensable, vu que tu fais déjà une emphase avec le `now`

Reply:
`a commonNames property remaining unchanged` => petite erreur grammaticale : les verbes en -ing sont pour une action qui est en cours, et ta propriété n'est pas en train d'être non changée :) 
il me semble que `an exact copy of commonNames property`, ou quelque chose de ce genre, correspondrait plus :thinking: